apps/swirl-docs/src/components/Icons/IconGridItem.tsx

```diff
@@ -34,25 +38,41 @@ const IconGridItem: FunctionComponent<IconGridProps> = ({
           id={id}
           tabIndex={index === 0 ? 0 : -1}
           role={role}
-          aria-label={`${icon}-icon`}
+          aria-label={`${icon}`}
           ref={reference}
           href={`#${icons[icon]?.name}`}
-          className="flex flex-col justify-center items-center py-4 border-1 rounded-lg"
+          className={classNames(
+            "flex flex-col justify-center items-center gap-2",
+            "aspect-square",
+            "supports-[not(container-type:inline-size)]:min-w-[120px]",
+            "supports-[not(container-type:inline-size)]:max-w-[120px]",
+            "py-4 border-1 rounded-lg",
+            { "bg-background-hovered": isSelected }
+          )}
           onKeyDown={(event) => handleKeyDown(event)}
           onClick={() => handleTileClick(icons[icon]?.name)}
         >
           <i
-            className={`swirl-icons-${icons[icon]?.name}28 text-icon-strong`}
+            className={`swirl-icons-${icons[icon]?.name}28 text-icon-strong text-2xl`}
           ></i>
-          <span className="text-text-subdued">{icons[icon]?.name}</span>
+          <span
+            className={classNames(
+              "text-text-subdued font-normal text-font-size-sm",
+              "max-w-[calc(120px_-_16px)]",
+              "overflow-hidden",
+              "text-ellipsis"
+            )}
+          >
+            {icons[icon]?.name}
+          </span>
         </a>
       </DesktopView>
       <MobileView>
         <div role={role}>
           <a
             id={`${id}-popover-trigger`}
             tabIndex={index === 0 ? 0 : -1}
-            aria-label={`${icon}-icon`}
+            aria-label={`${icon}`}
```

Comment:
**question:** Is the button text sufficient here?

Reply:
i think we can remove it^^